client/src/utils/scripts/helpers.js

```diff
@@ -0,0 +1,20 @@
+import { recipesApi } from '@/api';
+import { redirect } from '../router/routing';
+
+export async function deleteRecipe(recipeId) {
+  if (recipeId) {
+    await recipesApi.deleteRecipe({ recipeId });
+    redirect('UserRecipes');
+  }
+}
+
+export async function togglePublish(recipeId, isPublished) {
```

Comment:
again, this needs to be part of the component, why move it here if you use it in only one component. That might also have an effect on reactivity of your arrays